packages/ui/src/components/SelectInputV2/Dropdown.tsx

```diff
@@ -220,14 +220,23 @@ const handleClickOutside = (
   refSelect: RefObject<HTMLDivElement>,
   onSearch: Dispatch<SetStateAction<DataType>>,
   options: DataType,
+  isInsideModal: boolean,
 ) => {
   if (
     ref.current &&
     !ref.current.contains(event.target as Node) &&
     !refSelect.current?.contains(event.target as Node)
   ) {
-    setIsDropdownVisibile(false) // hide dropdown when clicking outside of the dropdown
-    onSearch(options) // reset displayed options to default when dropdown is hidden
+    if (isInsideModal) {
+      // Timeout when the selectInput is inside a modal to not close both the modal and the dropdown if it scrolls
+      setTimeout(() => {
+        setIsDropdownVisibile(false) // hide dropdown when clicking outside of the dropdown
+        onSearch(options) // reset displayed options to default when dropdown is hidden
+      }, 100)
+    } else {
+      setIsDropdownVisibile(false)
+      onSearch(options)
+    }
```

Comment:
haha no this we absolutely don't want. The issue with this is probably due to the modal itself more than the select input.